compiler/src/iree/compiler/Codegen/TransformDialectStrategies/GPU/MatmulTensorCoreStrategy.cpp

```diff
@@ -285,12 +294,15 @@ void iree_compiler::gpu::buildMatmulTensorCoreStrategy(
   }
 
   // Step 13. Late lowerings and cleanups.
-  // TODO: not a functional style op to avoid invalidating artificially.
-  funcH = b.create<transform::LowerMasksOp>(
-      pdl::OperationType::get(b.getContext()), funcH);
-  // TODO: not a functional style op to avoid invalidating artificially.
-  funcH = b.create<transform::MaterializeMasksOp>(
-      pdl::OperationType::get(b.getContext()), funcH);
+  if (!strategy.alignedLhs() || !strategy.alignedRhs() ||
```

Comment:
Are these guards actually necessary?
This should do nothing in case there are no masks right ?
If possible, I'd rather keep simplicity here.